### PR TITLE
WIP: DIG-350 | Event Box.

### PR DESCRIPTION
**Pull request by szatom**

### CODE REVIEW

File: web/modules/custom/diginomica_eventbox/src/Plugin/Block/EventBoxesBlock.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      ->execute();
+
+    $eventBoxes = $storage->loadMultiple($ids);
+
+    /* @var \Drupal\diginomica_eventbox\EventBoxInterface $eventBox */
+    foreach ($eventBoxes as $eventBox) {
+      $tid = 493;
+      $args = [$tid];
+
+      $view = Views::getView('event_articles');
```
Comment (dj1999) [inline web/modules/custom/diginomica_eventbox/src/Plugin/Block/EventBoxesBlock.php]: Maybe this is better to defining out of foreach one time.

File: web/modules/custom/diginomica_eventbox/src/Plugin/Block/EventBoxesBlock.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        $view->setArguments($args);
+        $view->setItemsPerPage($eventBox->getItemNum());
+        $view->setDisplay('default');
+        $view->preExecute();
+        $view->execute();
+        $render = $view->buildRenderable('default', $args, FALSE);
+
+        // @todo - add title to view.
+        $content[] = [
+          '#markup' => '<h3 class="view-title">' . $eventBox->getTitle() . '</h3>',
```
Comment (dj1999) [inline web/modules/custom/diginomica_eventbox/src/Plugin/Block/EventBoxesBlock.php]: Better if use #html_tag instead #markup.

File: web/modules/custom/diginomica_eventbox/src/Controller/EventBoxListBuilder.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    $header['layout'] = $this->t('Layout');
+    $header['weight'] = $this->t('Weight');
+    $header['status'] = $this->t('Status');
+    return $header + parent::buildHeader();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildRow(EntityInterface $entity) {
```
Comment (dj1999) [inline web/modules/custom/diginomica_eventbox/src/Controller/EventBoxListBuilder.php]: Use type EventBoxInterface instead EntityInterface.

File: web/modules/custom/diginomica_eventbox/src/Plugin/Block/EventBoxesBlock.php (hunk 2 above)
Comment (szatom) [reply web/modules/custom/diginomica_eventbox/src/Plugin/Block/EventBoxesBlock.php]: done

File: web/modules/custom/diginomica_eventbox/src/Plugin/Block/EventBoxesBlock.php (hunk 1 above)
Comment (szatom) [reply web/modules/custom/diginomica_eventbox/src/Plugin/Block/EventBoxesBlock.php]: Doesn't worked for me.

File: web/modules/custom/diginomica_eventbox/src/Controller/EventBoxListBuilder.php (hunk 3 above)
Comment (szatom) [reply web/modules/custom/diginomica_eventbox/src/Controller/EventBoxListBuilder.php]: Can't: we have to implement buildRow based on parent class's interface.
But I added:
`  public function buildRow(EntityInterface $entity) {
    /* @var $entity \Drupal\diginomica_eventbox\Entity\EventBox */`

**Comment by szatom** [comment]
Closed, this was part of master-DIG-354

**Pull request closed without merging**